reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerOperations.java

```diff
@@ -629,7 +643,26 @@ protected void onOutboundComplete() {
 			f = channel().writeAndFlush(newFullBodyMessage(EMPTY_BUFFER));
 		}
 		else if (markSentBody()) {
-			f = channel().writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT);
+			LastHttpContent lastHttpContent = LastHttpContent.EMPTY_LAST_CONTENT;
+			if (trailerHeadersConsumer != null && isTransferEncodingChunked(nettyResponse)) {
```

Comment:
there are two cases where any `Consumer` defined by the user simply gets silently ignored: `isTransferEncodingChunked` is false or `declaredHeaderNames == null`.

should these cases be documented in the javadoc? or should they lead to an exception? (especially if a `Consumer<TrailerHeaders>` is provided in the setup but no `declaredHeaderNames`/`HttpHeaderNames.TRAILER` entry is provided at all?)

Reply:
Please check it now, I added additional comments and javadoc